jetson/filter/src/__main__.py

```diff
@@ -283,16 +321,48 @@ def _runLKF(self):
         self.gps.fresh = False
         self.imu.fresh = False
 
+    def correct_bearing(self):
+        gps_bearing = self.gps.bearing.bearing_deg
+        imu_bearing = self.imu.bearing.bearing_deg
+
+        if time.time() - self.last_bearing_correction > self.config["Bearing_offset_interval_sec"]:
+            print("timeout passed")
+
+            if gps_bearing != 999:
+                print("valid gps bearing")
+
+                if self.nav_state == "Drive":
+                    print("drive state")
+
+                    if abs(self.drive_v_left - self.drive_v_right) < self.config["Drive_vel_diff_percent"]:
+                        print("driving straight")
+
+                        if abs(self.target_bearing - imu_bearing) < self.config["Bearing_error_threshold_deg"]:
+                            print("bearing error is low")
+
+                            # if abs(self.target_bearing_rate) < self.config["Bearing_rate_of_change_threshold_dps"]:
+                            #     print("bearing rate of change is high")
+
+                            old_bearing = imu_bearing + self.bearing_offset
+                            self.bearing_offset = gps_bearing - \
```

Comment:
imu bearing and gps bearing are same frame and same limit?? Are you sure you can do this?

Reply:
yeah this is definitely wrong because I'm never modding by 360, let me think about how to fix this